pysrc/bytewax/run.py

```diff
@@ -0,0 +1,237 @@
+    if ext == ".py":
+        path = fname
+
+    if os.path.basename(path) == "__init__":
+        path = os.path.dirname(path)
+
+    module_name = []
+
+    # move up until outside package structure (no __init__.py)
+    while True:
+        path, name = os.path.split(path)
+        module_name.append(name)
+
+        if not os.path.exists(os.path.join(path, "__init__.py")):
+            break
+
+    if sys.path[0] != path:
+        sys.path.insert(0, path)
+
+    return ".".join(module_name[::-1]) + f":{flow_name}"
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser(
+        prog="python -m bytewax.run", description="Run a bytewax dataflow"
+    )
+    parser.add_argument(
+        "import_str",
+        type=str,
+        help="Dataflow import string in the formats:\n"
+        "<module_name>:<dataflow_variable_name_or_factory_function>\n"
+        "<module_name>:<dataflow_factory_function>:<string_argument_for_factory>\n"
+        "Example: 'src.dataflow:flow' or 'src.dataflow:get_flow:string_argument'",
+    )
+    scaling = parser.add_argument_group(
+        "Scaling",
+        "You should use either '-p' to spawn multiple processes "
+        "on this same machine, or '-i/-a' to spawn a single process "
+        "on different machines",
+    )
+    scaling.add_argument(
+        "-p",
+        "--processes",
+        type=int,
+        help="Number of separate processes to run",
+        action=EnvDefault,
+        envvar="BYTEWAX_PROCESSES",
+    )
+    scaling.add_argument(
+        "-w",
+        "--workers-per-process",
+        type=int,
+        help="Number of workers for each process",
+        action=EnvDefault,
+        envvar="BYTEWAX_WORKERS_PER_PROCESS",
+    )
+    scaling.add_argument(
+        "-i",
+        "--process-id",
+        type=int,
+        help="Process id",
+        action=EnvDefault,
+        envvar="BYTEWAX_PROCESS_ID",
+    )
+    scaling.add_argument(
+        "-a",
+        "--addresses",
+        help="Addresses of other processes, separated by semicolumn:\n"
+        '-a "localhost:2021;localhost:2022;localhost:2023" ',
+        action=EnvDefault,
+        envvar="BYTEWAX_ADDRESSES",
+    )
+
+    # Config options for recovery
+    recovery = parser.add_argument_group("Recovery")
+    recovery.add_argument(
+        "--sqlite-directory",
+        type=pathlib.Path,
+        help="Passing this argument enables sqlite recovery in the specified folder",
+        action=EnvDefault,
+        envvar="BYTEWAX_SQLITE_DIRECTORY",
+    )
+    recovery.add_argument(
+        "--epoch-interval",
+        type=int,
+        default=10,
+        help="Number of seconds between state snapshots",
+        action=EnvDefault,
+        envvar="BYTEWAX_EPOCH_INTERVAL",
+    )
+
+    args = parser.parse_args()
+    args.import_str = _prepare_import(args.import_str)
+
+    # First of all check if a process_id was set with a different
+    # env var, used in the helm chart for deploy
```

Comment:
well, this is a custom way we use in the helmchart to define the proc id, which works by stripping the content of a variable from the content of another variable and parsing the resulting string to an int. It's ugly, and I wouldn't want users to do this, it's just a piece of code extracted from the old `parse_env`  for compatibility.

The other one is an env var reprsenting the hostfile path, rather than the addresses themselves. We want users to set the addresses in the variable, but the hostfile path ones was added for the same reason.

I guess at some point we might get rid of the old way of doing things, but Esteban asked me to not break everything for this release, so I added some compatibility workarounds.